airbyte-ci/connectors/regression-testing/src/regression_testing/main.py

```diff
@@ -0,0 +1,191 @@
+                dispatch(
+                    connector.container,
+                    FileBackend(f"{output_directory}/{connector.version}/{command}"),
+                    f"{output_directory}/{connector.version}",
+                    command,
+                    config,
+                    catalog,
+                    state,
+                ) for connector in [control_connector, target_connector]
+            ]
+        await asyncio.gather(*tasks)
+
+        await DiffComparator(client, output_directory).compare(control_connector, target_connector)
+
+
+async def dispatch(
+    container: dagger.Container,
+    backend: BaseBackend,
+    output_directory: str,
+    command: str,
+    config: Optional[SecretDict],
+    catalog: Optional[ConfiguredAirbyteCatalog],
+    state: Optional[Dict],
+):
+    runner = ConnectorRunner(container, backend, f"{output_directory}/{command}")
+
+    if command == "check":
+        await runner.call_check(config)
+
+    elif command == "discover":
+        await runner.call_discover(config)
+
+    elif command == "read":
+        runner = ConnectorRunner(container, backend, f"{output_directory}/read")
+        await runner.call_read(config, catalog)
+
+    elif command == "read-with-state":
+        runner = ConnectorRunner(container, backend, f"{output_directory}/read-with-state")
+        await runner.call_read_with_state(config, catalog, state)
+
+    elif command == "spec":
+        runner = ConnectorRunner(container, backend, f"{output_directory}/spec")
+        await runner.call_spec()
+
+    else:
+        raise NotImplementedError(f"{command} is not recognized. Must be one of {', '.join(COMMANDS)}")
+
+
+@click.command()
+@click.option(
+    "--connector-name",
+    help=(
+        "\"Technical\" name of the connector to be tested (e.g. `source-faker`)."
+    ),
+    default="latest",
+    type=str,
+)
+@click.option(
+    "--control-image-name",
+    help=(
+        "Control version of the connector to be tested. Records will be downloaded from this container and used as expected records for the target version."
+    ),
+    default="latest",
+    type=str,
+)
+@click.option(
+    "--target-image-name",
+    help=("Target version of the connector being tested."),
+    default="dev",
+    type=str,
+)
+@click.option(
+    "--output-directory",
+    help=("Directory in which connector output and test results should be stored."),
+    default="/tmp/test_output",
+    type=str,
+)
+@click.option(
+    "--command",
+    help=("Airbyte command."),
+    default="all",
+    type=click.Choice(COMMANDS + ["all"]),
+    required=True
+)
+@click.option(
+    "--config-path",
+    help=("Path to the connector config."),
+    default=None,
+    type=str,
+)
+@click.option(
+    "--catalog-path",
+    help=("Path to the connector catalog."),
+    default=None,
+    type=str,
+)
+@click.option(
+    "--state-path",
+    help=("Path to the connector state."),
+    default=None,
+    type=str,
+)
+def main(
```

Comment:
I think we should use `asynclick` instead of `click` so that we can declare `async def main` and just await on `_main`

Example of the same pattern [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/connectors_qa/src/connectors_qa/cli.py#L6) on the QA checks side.